src/components/Card.js

```diff
@@ -15,26 +25,43 @@ class Card {
     return cardElement;
   }
 
-  _handleLike = () => {
+  isLiked() {
+    return this._isLiked;
+  }
+
+  handleLike = () => {
     this._likeButton.classList.toggle("gallery__card-like_active");
   };
 
-  _handleDelete = () => {
+  handleDelete = () => {
     this._element.remove();
+    this._element = null;
   };
 
+  updateLikeView() {
+    if (this._isLiked) {
+      this._likeButton.classList.add("gallery__card-like_active");
+    } else {
+      this._likeButton.classList.remove("gallery__card-like_active");
+    }
+  }
+
   _setEventListeners() {
-    this._likeButton.addEventListener("click", this._handleLike);
-    this._cardTrash.addEventListener("click", this._handleDelete);
+    this._cardTrash.addEventListener("click", () => {
+      this._handleDeleteClick(this);
+    });
+    this._likeButton.addEventListener("click", () => {
+      this._handleLikeClick(this);
+    });
     this._cardImageEl.addEventListener("click", () => {
       this._handleImageClick({ name: this._name, link: this._link });
     });
   }
 
   generateCard() {
-    // select the like button, save to this
     this._element = this._getTemplate();
     this._likeButton = this._element.querySelector(".gallery__card-like");
+    this._deleteFormButton = document.querySelector("#delete-card-button");
```

Comment:
`    this._deleteFormButton = document.querySelector("#delete-card-button");
`

This is not needed here anymore